### PR TITLE
Add support for scala 2.11

### LINKED ISSUE #59: Add support for scala 2.11
I tried to use tapir with scala 2.11 but cannot locate any public jar files for that version of Scala. Would you be able to support this?

### PR DESCRIPTION
**Pull request by ghostbuster91**
This closes #59 

Solved issues:
- knownDirectSubclasses of <class> observed before subclass <class> registered - circe/circe#639
- http4s compilation errors -  enabling YpartialUnification
- other compilation errors - enabling Xexperimental
- fromResource is not a member of Source class - https://stackoverflow.com/questions/27360977/how-to-read-files-from-resources-folder-in-scala
- downgraded scala-logging due to missing artifact for scala 2.11 - https://github.com/lightbend/scala-logging/issues/160

### CODE REVIEW

**Comment by ghostbuster91** [comment]
I cannot merge with master because sbt is unable to resolve ` com.softwaremill:magnolia_2.11:0.11.0-sml:` Have you released it for 2.11? :)

**Comment by adamw** [comment]
@ghostbuster91 Released now, should be in central in 15-30 mins

File: build.sbt [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
 lazy val commonSettings = commonSmlBuildSettings ++ ossPublishSettings ++ Seq(
   organization := "com.softwaremill.tapir",
-  scalaVersion := "2.12.8",
-  scalafmtOnCompile := true
+  scalaVersion := scala2_12,
+  scalafmtOnCompile := true,
+  crossScalaVersions := Seq(scala2_11, scala2_12),
+  scalacOptions ++= 
+    (CrossVersion.partialVersion(scalaVersion.value) match {
+      case Some((2, v)) if v <= 12 =>
```
Comment (adamw) [inline build.sbt]: `if v == 11`?

File: build.sbt [hunk 2]
```diff
@@ -13,7 +26,7 @@ val sttpVersion = "1.5.12"
 lazy val loggerDependencies = Seq(
   "ch.qos.logback" % "logback-classic" % "1.2.3",
   "ch.qos.logback" % "logback-core" % "1.2.3",
-  "com.typesafe.scala-logging" %% "scala-logging" % "3.9.2"
+  "com.typesafe.scala-logging" %% "scala-logging" % "3.9.0"
```
Comment (adamw) [inline build.sbt]: why the downgrade?

File: docs/openapi-docs/src/test/scala/tapir/docs/openapi/VerifyYamlTest.scala [hunk 3]
```diff
@@ -274,8 +274,6 @@ case class Person(name: String, age: Int) extends Entity
 case class Organization(name: String) extends Entity
 
 sealed trait ErrorInfo
-object ErrorInfo {
-  case class NotFound(what: String) extends ErrorInfo
-  case class Unauthorized(realm: String) extends ErrorInfo
-  case class Unknown(code: Int, msg: String) extends ErrorInfo
-}
+case class NotFound(what: String) extends ErrorInfo
```
Comment (adamw) [inline docs/openapi-docs/src/test/scala/tapir/docs/openapi/VerifyYamlTest.scala]: anything wrong with nested classes and 2.11?

File: build.sbt (hunk 2 above)
Comment (ghostbuster91) [reply build.sbt]: I updated PR's description

File: docs/openapi-docs/src/test/scala/tapir/docs/openapi/VerifyYamlTest.scala (hunk 3 above)
Comment (ghostbuster91) [reply docs/openapi-docs/src/test/scala/tapir/docs/openapi/VerifyYamlTest.scala]: Sometimes, this is related to the first issue I mention in description.

**Pull request merged**